packages/expo-sqlite/ios/SQLiteModule.swift

```diff
@@ -371,6 +371,11 @@ public final class SQLiteModule: Module {
     try maybeThrowForClosedDatabase(database)
     try maybeThrowForFinalizedStatement(statement)
 
+    // The statement with parameter bindings is stateful,
+    // we have to guard with a critical section for thread safety.
+    statement.lock.lock()
+    defer { statement.lock.unlock() }
+
```

Comment:
```suggestion
 statement.semaphore.wait()
  defer { 
      statement.semaphore.signal() 
  }
```
There can be issues with thread safety with `NSLock` but it's up to you.